app/models.py

```diff
@@ -12,6 +14,42 @@ class FlathubUser(Base):
 
     id = Column(Integer, primary_key=True)
     display_name = Column(String)
+    deleted = Column(Boolean, nullable=False, default=False)
+    TABLES_FOR_DELETE = []
+
+    @staticmethod
+    def generate_token(db, user) -> str:
+        """
+        Generate a token which represents this user's state.  This token will be needed
+        when attempting to delete the user.
+        """
+        hasher = utils.Hasher()
+        # Add user info to be hashed
+        hasher.add_number(user.id)
+        hasher.add_string(user.display_name)
+        # Delete hash
+        for table in user.TABLES_FOR_DELETE:
+            table.delete_hash(hasher, db, user)
+        return hasher.hash()
+
+    @staticmethod
+    def delete_user(db, user, token: str) -> dict:
+        """
+        Attempt to delete the given user, we expect the user's token to match, otherwise
+        we will return an error.
+        """
+        current_token = FlathubUser.generate_token(db, user)
+        if token != current_token:
+            return {"status": "error", "error": "token mismatch"}
+
+        for table in user.TABLES_FOR_DELETE:
+            table.delete_user(db, user)
+
```

Comment:
It looks like you've forgotten to mark this user as deleted here.

Reply:
I think my changes address this.